extensions/wikia/SharedUserrights/SharedUserrights.class.php

```diff
@@ -68,53 +76,72 @@ public static function onUserLoadGroups( User $user ) {
 		return true;
 	}
 
-	static function addGlobalGroup( $user, $group ) {
-		global $wgExternalSharedDB, $wgWikiaGlobalUserGroups;
+	/**
+	 * @param User $user
+	 * @param $group
+	 * @return bool false, it's a hook
+	 */
+	static function addGlobalGroup( User $user, $group ) {
+		global $wgWikiaGlobalUserGroups;
 
 		if ( !in_array( $group, $wgWikiaGlobalUserGroups ) ) {
 			return true;
 		}
 
-		$dbw = wfGetDB( DB_MASTER, array(), $wgExternalSharedDB );
-		if( $user->getId() ) {
+		$dbw = self::getDB( DB_MASTER );
+		if ( $user->getId() ) {
 			$dbw->insert( 'user_groups',
-					array(
-						'ug_user'  => $user->getID(),
-						'ug_group' => $group,
-					     ),
-					__METHOD__,
-					array( 'IGNORE' ) );
+				[
+					'ug_user'  => $user->getID(),
+					'ug_group' => $group,
+				 ],
+				__METHOD__
+			);
 		}
 
-		wfRunHooks( 'AfterUserAddGlobalGroup', array( $user, $group ) );
+		global $wgMemc;
+		$wgMemc->delete( self::getMemcKey( $user ) );
```

Comment:
may be better to use WIkiaDataAccess::cachePurge() just in case some time in the future the actual key in memcached will no longer be exactly the same as the supplied one